pkg/common/gatewayapi_event_mappers.go

```diff
@@ -0,0 +1,49 @@
+package common
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	gatewayapiv1 "sigs.k8s.io/gateway-api/apis/v1"
+)
+
+// HTTPRouteToParentGatewaysEventMapper is an EventHandler that maps HTTPRoute events to gateway events,
+// by going through the parentRefs of the route
+type HTTPRouteToParentGatewaysEventMapper struct {
+	Logger logr.Logger
+}
+
+func (m *HTTPRouteToParentGatewaysEventMapper) Map(_ context.Context, obj client.Object) []reconcile.Request {
```

Comment:
Except for the type it returns (`[]reconcile.Request` vs `[]string`), isn't this function practically the same as the one defined by `controllers.addHTTPRouteByGatewayIndexer` for the list filtering?